akka-actor-tests/src/test/scala/akka/io/TcpConnectionSpec.scala

```diff
@@ -911,6 +911,7 @@ class TcpConnectionSpec extends AkkaSpec("""
       new ChannelRegistration {
         def enableInterest(op: Int): Unit = interestCallReceiver.ref ! op
         def disableInterest(op: Int): Unit = interestCallReceiver.ref ! -op
+        def wakeUp(): Unit = ()
```

Comment:
> Hmm the selection is already cancelled automatically as a side effect of closing the channel though, so it's not really needed to do it here explicitly ourselves

Yes, I know, it's more about the API. It somehow seems more reasonable to request (or report) cancellation than just to wake up the selector which should be an implementation detail.